pkg/plugins/resources/hcl/spec.go

```diff
@@ -0,0 +1,103 @@
+package hcl
+
+import (
+	"errors"
+
+	"github.com/sirupsen/logrus"
+)
+
+/*
+"hcl"  defines the specification for manipulating "hcl" files.
+It can be used as a "source", a "condition", or a "target".
+*/
+type Spec struct {
+	/*
+		"file" defines the hcl file path to interact with.
+
+		compatible:
+			* source
+			* condition
+			* target
+
+		remark:
+			* "file" and "files" are mutually exclusive
+			* protocols "https://", "http://", and "file://" are supported in path for source and condition
+	*/
+	File string `yaml:",omitempty"`
+	/*
+		"files" defines the list of hcl files path to interact with.
+
+		compatible:
+			* source
+			* condition
+			* target
+
+		remark:
+			* file and files are mutually exclusive
+			* when using as a source only one file is supported
+			* protocols "https://", "http://", and "file://" are supported in file path for source and condition
+	*/
+	Files []string `yaml:",omitempty"`
+	/*
+		"key" defines the hcl attribute path.
+
+		compatible:
+			* source
+			* condition
+			* target
+
+		example:
+			* key: resource.aws_instance.app_server.ami
+			* key: resource.helm_release.prometheus.version
+			* key: plugin.aws.version
+
+	*/
+	Key string `yaml:",omitempty"`
```

Comment:
@olblak thoughts on the name `Key`? I've gone with it as it aligns with `yaml`, but could be convinced on alternative.

Reply:
I am like you, I don't have strong opinions.

In order plugin I used:
* xml - `path` for xamlpath
* json - `key` to align on yaml (but it accepts jsonpath) but there is also `query for more advanced dasel queries`
* yaml - `key` for historical reason

I am familiar with key so :shrug:

Reply:
Usually I rely on @hervelemeur to be more picky :D

Reply:
`Key` doesn't feel quite natural (especially if its not common across the board).

`Path` or `Address` feels closer to correct. I normally rely on @quotidian-ennui for opinions like this.

`hcledit` itself uses `Address` 

```text
hcledit attribute set --help
Set a value of matched attribute at a given address

Arguments:
  ADDRESS          An address of attribute to set.
  VALUE            A new value of attribute.
```

Reply:
Thanks for the detailed explanation.
I think I would prefer `path` for jsonpath, xmlpath, yamlpath,etc. but `address` is fine to me as well

We could definitely consider deprecating the `key` usage in other plugin like `yaml` or `json` in favor of `path` to be more consistent (in a different pullrequest)

Something worth to keep in mind, up to now we manage to not introduce (voluntarily) breaking changes but instead deprecation.

Let see what other thing about this

Reply:
We could have it as `attribute` because that's the intention (we are editing an attribute in the hcl file).

In the future there may be a desire to change a _block_ on in updatecli (it's possible but unlikely) which means that address would need to be disambiguated with "PARENT_ADDRESS" & "CHILD_ADDRESS" (from hcledit block append -h)

Although at that point i suspect you might have nested configuration (much your `changedif` example above).

Which is a roundabout of saying: I have an opinion (attribute) but I don't care enough to argue for it either way 😉

Reply:
Thanks for the feedback, I'll open an issue to keep track that we plan to standardise to "path"